subprojects/packagefiles/libyaml/meson.build

```diff
@@ -0,0 +1,54 @@
+project(
+  'yaml-0.1', 'c',
+  license: 'MIT',
+  meson_version: '>=0.49.2',
+  version: '0.2.5',
+)
+
+pkg = import('pkgconfig')
+
+add_project_arguments('-DHAVE_CONFIG_H=1', language: 'c')
+
+yaml_version = meson.project_version()
+yaml_major_version = yaml_version.split('.')[0]
+yaml_minor_version = yaml_version.split('.')[1]
+yaml_patch_version = yaml_version.split('.')[2]
+cdata = configuration_data({
+  'YAML_VERSION_MAJOR' : yaml_major_version,
+  'YAML_VERSION_MINOR' : yaml_minor_version,
+  'YAML_VERSION_PATCH' : yaml_patch_version,
+})
+cdata.set_quoted('YAML_VERSION_STRING', meson.project_version())
+
+config_h = configure_file(
+  output: 'config.h',
+  configuration: cdata)
+
+yaml_lib = library('yaml',
```

Comment:
This is wrong. :(

Actually, this is confusing as heck.

`src/Makefile.am` passes the `-release $(YAML_LT_RELEASE)` flag to libtool, producing a library named `yaml-0` and linked to via... `-lyaml`

Reply:
This cannot be replicated in meson. The soname (ABI experience) is named `yaml-0`, while the pkg-config file exposes `yaml`.

The perfect is the enemy of the good -- so let's merge this using `yaml-0`. This is compatible in the soname table, and even generates non-equal but still *compatible* pkg-config files, and only breaks people using the equivalent of `cc.find_library('yaml')`. At this point we can say "you probably should build with autotools if you are doing system packaging.

Can you leave a comment to the effect of "this cannot avoid breaking the use of `-lyaml` due to strange libtool magic, but everything else works" so that no one forgets why this is happening? If upstream is ever interested in meson, they can debate the issue at that time.

Reply:
done